pkg/channel/channel.go

```diff
@@ -36,3 +36,8 @@ type AppChannel interface {
 	HealthProbe(ctx context.Context) (bool, error)
 	SetAppHealth(ah *apphealth.AppHealth)
 }
+
+// HTTPEndpointAppChannel is an abstraction over communications with http endpoint resources.
+type HTTPEndpointAppChannel interface {
```

Comment:
Right now I'd opt to keep it consistent with AppChannel, in the future we can track work to make this change for both if it can actually account for both.